src/com/maddyhome/idea/vim/extension/entiretextobj/VimEntireTextObjExtension.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * IdeaVim - Vim emulator for IDEs based on the IntelliJ platform
+ * Copyright (C) 2003-2020 The IdeaVim authors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.maddyhome.idea.vim.extension.entiretextobj;
+
+import com.intellij.openapi.actionSystem.DataContext;
+import com.intellij.openapi.editor.Caret;
+import com.intellij.openapi.editor.Editor;
+import com.maddyhome.idea.vim.command.*;
+import com.maddyhome.idea.vim.common.TextRange;
+import com.maddyhome.idea.vim.extension.VimExtension;
+import com.maddyhome.idea.vim.extension.VimExtensionHandler;
+import com.maddyhome.idea.vim.handler.TextObjectActionHandler;
+import com.maddyhome.idea.vim.listener.SelectionVimListenerSuppressor;
+import com.maddyhome.idea.vim.listener.VimListenerSuppressor;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.EnumSet;
+
+import static com.maddyhome.idea.vim.extension.VimExtensionFacade.putExtensionHandlerMapping;
+import static com.maddyhome.idea.vim.extension.VimExtensionFacade.putKeyMapping;
+import static com.maddyhome.idea.vim.extension.entiretextobj.VimEntireTextObjExtension.EntireHandler.*;
+import static com.maddyhome.idea.vim.group.visual.VisualGroupKt.vimSetSelection;
+import static com.maddyhome.idea.vim.helper.StringHelper.parseKeys;
+import static java.util.Collections.emptyList;
+
+/**
+ * Port of vim-entire:
+ * https://github.com/kana/vim-textobj-entire
+ *
+ * <p>
+ * vim-textobj-entire provides two text objects:
+ * <ul>
+ *   <li>ae targets the entire content of the current buffer.</li>
+ *   <li>ie is similar to ae, but ie does not include leading and trailing empty lines. ie is handy for some situations. For example,</li>
+ *   <ul>
+ *     <li>Paste some text into a new buffer (<C-w>n"*P) -- note that the initial empty line is left as the last line.</li>
+ *     <li>Edit the text (:%s/foo/bar/g etc)</li>
+ *     <li>Then copy the resulting text to another application ("*yie)</li>
+ *   </ul>
+ * </ul>
+ *
+ * See also the reference manual for more details at:
+ * https://github.com/kana/vim-textobj-entire/blob/master/doc/textobj-entire.txt
+ *
+ * @author Alexandre Grison (@agrison)
+ */
+public class VimEntireTextObjExtension implements VimExtension {
+  @Override
+  public @NotNull
+  String getName() {
+    return "entiretextobj";
```

Comment:
Seems like the plugin is called `vim-textobj-entire`. Should we name this one as`textobj-entire`?

Reply:
Should totally be named `textobj-entire`, I'll rename it also in the `README`